### ISSUE TITLE
Minimized Windows

### ISSUE THREAD

**Issue by steventheworker** [steventheworker/DockAltTab#8]
Default MacOS behavior is to open the app on single mouse click if only one window of that app exists. With DockAltTab, this requires either: hover & select window or click dock icon or double click dock icon

**Comment by sdburrows** [comment]
I am glad this issue was brought up. I was about to create one myself. I noticed the click to minimise/maximise behaviour doesn't work sometimes. For e.g. a maximised windows, it would sometimes requires two clicks to minimise it. Also, there is a delay when clicking on minimised window to maximise.

**Comment by steventheworker** [comment]
In your example do you mean Full Screen or Maximized via an app like Rectangle? And by "delay when clicking on minimized window" do you mean the AltTab preview button to full screen?

**Comment by sdburrows** [comment]
I should be clearer with my explanations. When enabling the option "Modifying click behaviour" within DockAltTab, I expected the dock to behave like Windows taskbar. Instead, the behaviour is a bit erratic. The steps I used to reproduce this is: 
1. Open an app / window
2. Minimise it using the yellow "-" button
3. Click on dock app icon to restore it (not the preview window), nothing happens
4. Click again and the window will open and hide again

The expected behaviour in Step 3 is the window should reappear. 

I wonder, rather than click dock icon to hide, how about click to minimise, same behaviour as the "-" yellow  symbol on the top left window. Would this solve the issue? 

The delay is possibly related to hiding window than minimise window.

**Comment by steventheworker** [comment]
Yes this sounds like a combination of a different issue (minimized windows) + issue 11 (trackpad tap clicks). So

**Comment by steventheworker** [comment]
Yes this sounds like a combination of a different issue (minimized windows) + issue 11 (trackpad tap clicks).

**Comment by Moolynx** [comment]
> EDIT: yes your theory is correct, and minimizing > hiding would solve this, but only if ALL windows were minimized on click / I have DockAltTab not react to clicks on dock icons with >1 previews.

Does this mean you will look into replacing the action from click to hide to click to minimise instead? 

Also, why would you disable DockAltTab from reacting to clicks on dock icons when there is more than 1 preview window?

**Comment by WynStaXxx** [comment]
> Also, why would you disable DockAltTab from reacting to clicks on dock icons when there is more than 1 preview window?

I think that was done so it doesnt auto select the app, what it the default macos behavior when click the icon. When you  have >2+ previews you can now select which preview otherwise you wouldn't be able to have that option. BTW he is talking about icon not the previews.

**Comment by steventheworker** [comment]
This is fixed in v2.91.0 ... I think!
Gonna keep this issue open for a while longer just in case you guys have any input.
